RFCs/FS-1142-Extended-numeric-literal.md

```diff
@@ -0,0 +1,137 @@
+
+This RFC covers the detailed proposal for this suggestion.
+
+- [x] Suggestion: [Underscores in numeric literals after prefix and before suffix](https://github.com/fsharp/fslang-suggestions/issues/718), [Extend custom numeric types to support floating point literals](https://github.com/fsharp/fslang-suggestions/issues/445) and [Hexadecimal, octal and binary custom numeric literals](https://github.com/fsharp/fslang-suggestions/issues/754)
+- [x] Approved in principle
+- [ ] [Implementation](https://github.com/dotnet/fsharp/pull/17242)
+- [ ] Design Review Meeting(s) with @dsyme and others invitees
+- [Discussion](https://github.com/fsharp/fslang-design/discussions/769)
+
+# Summary
+
+This RFC will allow the following things:
+
+- Underscores in numeric literals after prefix and before suffix like `0x_1` or `1_l` or mixed them up like `0x_1_l`.
+- Hexadecimal, octal, binary and floating point custom numeric literals like `0x1I` or `1.0G`
+
+# Motivation
+
+Make the language more consistent and easier to read. Enhance the custom numeric literals feature by supporting not only integers but also floats.
+
+# Detailed design
+
+1. Underscores can now be placed after prefix (`0x`, `0o`, `0b`) and before suffix (eg. `UL`, `y`, `m`, `f`). These are forbidden [previously](https://github.com/fsharp/fslang-design/blob/main/FSharp-4.1/FS-1005-underscores-in-numeric-literals.md#detailed-design).
+
+    ```fsharp
+    let pi1 = 3_.1415F      // Invalid cannot put underscores adjacent to a decimal point
+    let pi2 = 3._1415F      // Invalid cannot put underscores adjacent to a decimal point
+    let socialSecurityNumber1 = 999_99_9999_L         // OK (Invalid *previously*)
+
+    let x1 = _52              // This is an identifier, not a numeric literal
+    let x2 = 5_2              // OK (decimal literal)
+    let x3 = 52_              // Invalid cannot put underscores at the end of a literal
+    let x4 = 5_______2        // OK (decimal literal)
+
+    let x5 = 0_x52            // Invalid cannot put underscores in the 0x radix prefix
+    let x6 = 0x_52            // OK (Invalid *previously*)
+    let x7 = 0x5_2            // OK (hexadecimal literal)
+    let x8 = 0x52_            // Invalid cannot put underscores at the end of a number
+
+    // In contrast to Java, literals with leading zeros are decimal in F#.
+    let x9 = 0_52             // OK (decimal literal)
+    let x10 = 05_2            // OK (decimal literal)
+    let x11 = 052_            // Invalid cannot put underscores at the end of a number
+
+    // To create an octal literal, prefix it with '0o' similar to hexadezimal literals. The same rules apply:
```

Comment:
Keep in mind that RFC's are not tutorials ;). We can assume that anybody reading these has a deep knowledge of the subject :)
```suggestion
    // Octal literals
```